File: php/testcode/pathtraver_015.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_pt_symlink
function pathtraver_symlink(BenchmarkRequest $req): BenchmarkResponse {
    $target = $req->param('target');
    $link = "/var/www/links/" . basename($req->param('linkname'));
    symlink($target, $link); // vuln-code-snippet vuln-line php_pt_symlink
    return BenchmarkResponse::ok("Symlink created");
}
// vuln-code-snippet end php_pt_symlink

File: ruby/testcode/pathtraver_015.rb
require_relative 'shared'

# vuln-code-snippet start ruby_pt_readlines
def read_log(req)
  log_name = req.param('log')
  lines = File.readlines("/var/log/#{log_name}") # vuln-code-snippet vuln-line ruby_pt_readlines
  BenchmarkResponse.ok(lines.join)
end
# vuln-code-snippet end ruby_pt_readlines
